Add user id to medical record

NadiaArroyo25 opened the pull request:
add an id table for medical records so that only the users who create them can access them and you can still share the patients if necessary

NadiaArroyo25 committed:
Add user id to medical record

add an id table for medical records so that only the users who create them can access them and you can still share the patients if necessary

Ligator commented on db/migrate/20240507004300_add_user_id_to_medical_records.rb:
We will need to change in another PR this model (and table) name from `Usuario` to `User`.

NadiaArroyo25 replied:
That's what I thought :(

Ligator commented on db/migrate/20240507004300_add_user_id_to_medical_records.rb:
```diff
@@ -0,0 +1,10 @@
+class AddUserIdToMedicalRecords < ActiveRecord::Migration[7.1]
+  def change
+    add_reference :medical_records, :usuario, null: true, foreign_key: true, default: 1
```
This will work fine in your local environment but you don't know if there is a user with ID 1 in other environments like testing or production.
So you need to fin or create the user.
```suggestion
    user = User.first || User.create(email: "temporary_email@example.com", password: "pass", password_confirmation: "pass")
    add_reference :medical_records, :usuario, null: true, foreign_key: true, default: user.id
```

NadiaArroyo25 replied:
To do that, is it better to undo the migration and do it again? or modify my file and run the same migration?

Ligator replied:
Given this change won't make any difference in your local, you can just commit the change so it will run for your production and testing environments. Only for this case scenario.

AlfredoJA commented on app/controllers/medical_records_controller.rb:
```diff
@@ -3,9 +3,8 @@ class MedicalRecordsController < ApplicationController
   # GET /medical_records or /medical_records.json
   def index
-    @medical_records = MedicalRecord.where(patient_id: params[:patient_id]).all
+    @medical_records = MedicalRecord.where(patient_id: params[:patient_id], usuario_id: current_usuario.id)
```
If I'm understanding your relation maybe you can use this way. 

```suggestion
    @medical_records = current_usuario.medical_records.where(patient_id: params[:patient_id])
```

But I'm not sure if this is correct.

NadiaArroyo25 replied:
hmm with instance variable if it works

AlfredoJA replied:
Both works, but is more common use `current_usuario.medical_records`.

AlfredoJA commented on app/controllers/medical_records_controller.rb:
```diff
@@ -22,6 +21,7 @@ def edit
   # POST /medical_records or /medical_records.json
   def create
     @medical_record = MedicalRecord.new(medical_record_params)
+    @medical_record.usuario_id = current_usuario.id
```
I think this way is better:
```suggestion
    medical_record_params.merge!(usuario_id: current_usuario.id)
    @medical_record = MedicalRecord.new(medical_record_params)
```

AlfredoJA replied:
Also you can send the `usuario_id` as part of your form.

Ligator replied:
No, she cannot receive the `usuario_id` from scary internet, it should be set in the backend with the logged in user.

AlfredoJA replied:
You're right.

NadiaArroyo25 committed:
Update db/migrate/20240507004300_add_user_id_to_medical_records.rb

Co-authored-by: Abel Pacheco <abelpacheco@outlook.com>

NadiaArroyo25 committed:
Update app/controllers/medical_records_controller.rb

Co-authored-by: AlfredoJA <alfre.jangeles@hotmail.com>

NadiaArroyo25 committed:
update

Ligator commented on db/migrate/20240507004300_add_user_id_to_medical_records.rb:
```diff
@@ -0,0 +1,11 @@
+class AddUserIdToMedicalRecords < ActiveRecord::Migration[7.1]
+  def change
+    user = User.first || User.create(email: "temporary_email@example.com", password: "pass", password_confirmation: "pass")
```
Please ensure this line is working, I mean, maybe the password doesn't meet the requirements.

NadiaArroyo25 committed:
modify

Well, I already modified the migration, it didn't work at first, but not because of the password, it already worked, so I went back to my previous queries and it now works.
